src/pages/deadlines/DeadlinesPage.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { Box } from "native-base";
+import { DrawerScreenProps } from "@react-navigation/drawer";
+import { RootDrawerParamList } from "../../interfaces/navigatorInterfaces";
+import { CalendarList } from "react-native-calendars";
+import { useAppSelector } from "../../redux";
+import { isAssignment } from "../../utils/course";
+
+type Props = DrawerScreenProps<RootDrawerParamList, "Deadlines">;
+
+const DeadlinePage = ({ navigation }: Props) => {
+  const modules = useAppSelector((state) => state.modules);
+  const courses = useAppSelector((state) => state.courses);
+  const deadlines = modules.filter((mod) => isAssignment(mod.type));
+
+  const courseUrlToColors = {}
+  for(let i=0; i<courses.length; i++) {
+    courseUrlToColors[courses[i].courseUrl] = courses[i].courseColor
+  }
+
+  const dates = {}
```

Comment:
Typing for this as well. Will reduce TypeScript errors (even tho is okay for JS)